submissions/kt-std/friends-app/js/script.js

```diff
@@ -0,0 +1,338 @@
+	limitType,
+	limitInputIdValue,
+	limitTypeValue,
+	ageLimit
+) {
+	if (limitInputId === limitInputIdValue && limitType === limitTypeValue)
+		limitInput.value = ageLimit;
+}
+
+function getCertainAgeLimit(friendsArray, value) {
+	const sortedArray = friendsArray.sort((a, b) => a.age - b.age);
+	return value === "min"
+		? sortedArray[0].age
+		: sortedArray[sortedArray.length - 1].age;
+}
+
+function sortCards(e, friendsArray) {
+	if (e.target.classList.contains("list__item")) {
+		document.querySelector(".select__face-item").innerText =
+			e.target.textContent;
+		sortCardsArray(e.target.attributes.value.value, friendsArray);
+	}
+}
+
+function sortCardsArray(condition, friendsArray) {
+	switch (condition) {
+		case "ND":
+			FRIENDS_ARRAY.sort((a, b) => b.firstName.localeCompare(a.firstName));
+			break;
+		case "NA":
+			FRIENDS_ARRAY.sort((a, b) => a.firstName.localeCompare(b.firstName));
+			break;
+		case "AD":
+			FRIENDS_ARRAY.sort((a, b) => b.age - a.age);
+			break;
+		case "AA":
+			FRIENDS_ARRAY.sort((a, b) => a.age - b.age);
+			break;
+	}
+}
+
+function findSubstring(string, substring) {
+	return string.toLowerCase().indexOf(substring.toLowerCase()) >= 0
+		? true
+		: false;
+}
+
+function findMatchesWithPropertiesValues(
+	propertiesList,
+	friendsArray,
+	substring
+) {
+	return friendsArray.filter((friend) => {
+		return propertiesList
+			.map((property) => findSubstring(friend[property], substring))
+			.some((el) => el);
+	});
+}
+
+function filterByGenderValue(genderList, friendsArray) {
+	return friendsArray.filter((friend) =>
+		genderList.some((gender) => friend.gender === gender.value)
+	);
+}
+
+function filterByAgeLimits(minAge, maxAge, friendsArray) {
+	return friendsArray.filter(
+		(friend) => friend.age >= minAge && friend.age <= maxAge
+	);
+}
+
+function filterByGender(e, friendsArray) {
+	let checkboxes = document.querySelectorAll("input[type=checkbox]");
+	checkboxes = Array.from(checkboxes).filter((checkbox) => checkbox.checked);
+	return filterByGenderValue(checkboxes, friendsArray);
+}
+
+function filterByAge(friendsArray) {
+	const min = document.getElementById("minAge"),
+		max = document.getElementById("maxAge");
+	if (min.value && max.value) {
+		return filterByAgeLimits(min.value, max.value, friendsArray);
+	}
+}
+
+checkFiltersChanged = (event) =>
+	["list__item", "number", "checkbox"].some(
+		(filter) => event.target.className === filter
+	);
+
+function updateCards(event) {
+	if (checkFiltersChanged(event)) {
+		FRIENDS_ARRAY = INITIAL_FRIENDS_ARRAY;
+		sortCards(event, FRIENDS_ARRAY);
+		FRIENDS_ARRAY = filterByAge(FRIENDS_ARRAY);
+		FRIENDS_ARRAY = filterByGender(event, FRIENDS_ARRAY);
+		appendFriendsCards(FRIENDS_ARRAY);
+	}
+}
+
+document.querySelector("#showFiltersButton").addEventListener("click", (e) => {
+	const filtersContainer = document.querySelector(".filters__container");
+	filtersContainer.classList.toggle("display");
+	Array.from(filtersContainer.children).forEach((node) => {
```

Comment:
This logic should be done via CSS and not by JS

Reply:
I've left only the `toggle` method to show and hide the filters container. Other animation effects are now being done via CSS.